index.php

```diff
@@ -115,6 +116,13 @@ function failed($reason) {
 require($CFG->dirroot . '/lib/setup.php');
 require_once($CFG->libdir.'/filelib.php');
 
+//IP Locking, check for remote IP in validated list, if not, exit
+$allowedips = get_config('tool_heartbeat','ipconfig');
+if ((!(remoteip_in_list($allowedips))) && (trim($allowedips) !== '')){
+    print "IP is not in validated range";
```

Comment:
There is a core api for notications:

echo $OUTPUT->notification($errormsg);

Reply:
ah sorry you are already using this below in the gui, in this context I think what you did is good. Maybe the only improvement could be sending a http 403 instead